src/main/java/codeu/controller/SearchServlet.java

```diff
@@ -43,49 +32,26 @@ void setUserStore(UserStore userStore) {
   }
 
   /**
-   * Sets the ConversationStore used by this servlet. This function provides a 
-   * common setup method for use by the test framework or the servlet's init() function.
-   */
-  void setConversationStore(ConversationStore conversationStore) {
-    this.conversationStore = conversationStore;
-  }
-
-  /**
-   * Sets the EventStore used by this servlet. This function provides a common 
-   * setup method for use by the test framework or the servlet's init() function.
-   */
-  void setMessageStore(MessageStore messageStore) {
-    this.messageStore = messageStore;
-  }
-
-  /**
-   * TODO: Documentation
+   * This function gets the Users that contains the searchRequest (username) 
+   * and sets up the JSON object to be parsed. 
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) 
       throws IOException, ServletException {
     String searchRequest = (String)request.getParameter("searchRequest");
-    request.setAttribute("searchRequest", searchRequest);
-
-    List<User> users = userStore.getUsers(searchRequest);
-    request.setAttribute("users", users);
-
-    // List<Conversation> conversations = conversationStore.getAllConversations();
-    // request.setAttribute("conversations", conversations);
-
-    // List<Message> messages = messageStore.getAllMessages();
-    // request.setAttribute("messages", messages);
-
-    request.getRequestDispatcher("/WEB-INF/view/search.jsp").forward(request, response);
-  }
 
-  /**
-  * TODO: Determine if this is needed.
-  */
-  @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) 
-      throws IOException, ServletException {
-    // TODO
+    String json;
+    if (searchRequest == null) {
+      json = new Gson().toJson("[]");
+    }
+    else {
+      List<User> users = userStore.getUsers(searchRequest);
```

Comment:
If we make `getUsers` return an empty list when `searchRequest` is null, an empty string, or a non-matching string, then this simplifies this code block, it then just becomes:
`String json = new Gson().toJson(userStore.getUsers(searchRequest));`
It also makes sense that the return type of the function is consistent to always be a list.